mysql-test/suite/client/client-env-variable.test

```diff
@@ -0,0 +1,76 @@
+-- source include/have_log_bin.inc
+
+# Set up environment varibles for client programs
+# The environment variables for client programs have default options file
+# They interfere with the MTR test so I am creating new variables for them
+
+--let MYSQLADMIN = $MYSQL_BINDIR/client//mariadb-admin
+--let MYSQLBINLOG = $MYSQL_BINDIR/client//mariadb-binlog
+--let MYSQLCHECK = $MYSQL_BINDIR/client//mariadb-check
+--let MYSQLDUMP = $MYSQL_BINDIR/client//mariadb-dump
+--let MYSQLIMPORT = $MYSQL_BINDIR/client//mariadb-import
+--let MYSQLSHOW = $MYSQL_BINDIR/client//mariadb-show
+--let MYSQLSLAP = $MYSQL_BINDIR/client//mariadb-slap
+
+# Creating a table for the client programs
+USE test;
+CREATE TABLE pet (name VARCHAR(20));
+
+# Creating a data file for mysqlimport
+write_file $MYSQL_TMP_DIR/pet;
+buster
+bob
+EOF
+
+# Options for client program
+--let $options = --user=root --port=16000 --socket=$MYSQL_TMP_DIR/mysqld.1.sock
```

Comment:
`port` and `socket` aren't fixed.

`$MASTER_MYPORT` and `$MASTER_MYSOCK` (from mysql-test/include/default_my.cnf - no include required - its a default).

Or, if you create a `client-env-variable.cnf` like:
```
!include include/default_my.cnf

[ENV]
MARIADB_HOST=localhost
```

This is a true environment variable.

Then `$options` to be `--port=$MASTER_MYPORT` (opmit socket so it doesn't get used) and use a positive test (no more `--error`) to ensure the connection is there.

Reply:
> no more --error
I am not sure if you want to remove the negative test case with `--error`. I added the positive test case but kept the negative test case in the new revision.